src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilterTest.java

```diff
@@ -0,0 +1,485 @@
+
+        try {
+            filter.accept(auditEvent);
+            fail(IllegalStateException.class.getSimpleName() + " is expected");
+        }
+        catch (IllegalStateException ex) {
+            assertEquals("Invalid exception message",
+                "Cannot read source file: " + fileName, ex.getMessage());
+
+            final Throwable cause = ex.getCause();
+            assertTrue("Exception cause has invalid type",
+                cause instanceof FileNotFoundException);
+            assertEquals("Invalid exception message",
+                fileName + " (No such file or directory)", cause.getMessage());
+        }
+    }
+
+    @Test
+    public void testFilterWithCustomMessageFormat() throws Exception {
+        final DefaultConfiguration filterCfg =
+            createModuleConfig(SuppressWithPlainTextCommentFilter.class);
+        final String messageFormat =
+            ".*" + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB) + ".*";
+        filterCfg.addAttribute("messageFormat", messageFormat);
+
+        final DefaultConfiguration fileTabCheckCfg =
+            createModuleConfig(FileTabCharacterCheck.class);
+        fileTabCheckCfg.addAttribute("eachLine", "true");
+
+        final DefaultConfiguration regexpCheckCfg = createModuleConfig(RegexpSinglelineCheck.class);
+        regexpCheckCfg.addAttribute("id", "ignore");
+        regexpCheckCfg.addAttribute("format", ".*[a-zA-Z][0-9].*");
+
+        final String[] suppressed = {
+            "8:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB),
+        };
+
+        final String[] violationMessages = {
+            "6: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "8:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB),
+            "8: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "10: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "13: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+        };
+
+        verify(
+            createCustomChecker(filterCfg, fileTabCheckCfg, regexpCheckCfg),
+            getPath("InputSuppressWithPlainTextCommentFilterCustomMessageFormat.java"),
+            removeSuppressed(violationMessages, suppressed)
+        );
+    }
+
+    private Checker createCustomChecker(Configuration... childConfigs) throws Exception {
+        final DefaultConfiguration checkerCfg = createModuleConfig(Checker.class);
+        Arrays.stream(childConfigs).forEach(checkerCfg::addChild);
+        return createChecker(checkerCfg);
+    }
+
+    @Override
+    protected DefaultConfiguration createRootConfig(Configuration config) {
```

Comment:
@MEZk creating custom root configurations is now forbidden.
Please see https://github.com/checkstyle/checkstyle/pull/5277 for similar tests from where this was copied.

Reply:
@rnveach 
```createRootConfig```  is declared as ```protected``` in master. Protected, public methods are supposed to be overridden. The method in ```AbstractModuleTestSupport``` does not have any documentation which explains why it is prohibited to be overridden. ```createRootConfig``` is not marked with ```final``` modifier. Thus, it means that if the developer is not satisfied with the logic, provided by the abstract (parent) class, he have the right to override the method. The last but not the least, once again, what is the reason to copy paste the code which creates the root config with multiple children in the loop in each test?

Judging by the [comment](https://github.com/checkstyle/regression-tool/pull/85#issuecomment-325333038) 

> 6. Some UTs don't use createModuleConfig

Why it is so special? If it is required for the regression tool, I still do not see any ```final``` modifier to think of the method as forbidden to be overridden.

Reply:
> Protected, public methods are supposed to be overridden.
>  I still do not see any final modifier to think of the method as forbidden to be overridden.

Yes, this was never done. I will make a new PR with this change.

> Test should be as clean and easy to read as possible.

Seeing a common method like `createRootConfig` and finding out it does alot of extra stuff does not make the test easier imo.

> what is the reason to copy paste the code which creates the root config with multiple children in the loop in each test?

Regression tool will only look at the methods marked with `@Test` to prevent false positives, so the heart of what we pick up must be in the main test.

If you have need for multiple tests to use alot of common children configurations, you can move all out into a custom verify method. See https://github.com/checkstyle/checkstyle/blob/8441fcac4471d499fff0b5bf9141175b78fdb4d7/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java#L268 which is similar to the area you copied your tests from.
You just can't move the test subject of the class outside the method as this will hide it's property values from the regression tool.

Reply:
@rnveach 
> If you have need for multiple tests to use alot of common children configurations, you can move all out into a custom verify method.

1). Explain what is the difference between the method which creates custom checker configuration with multiple children and custom verify method which will contain the logic which creates the checker with multiple children? What is the point?
2). I still need the checker instance which configuration consists of more than one child. I do not want to use ```createChecker``` methods, as they do not correspond to my needs. ```verifySuppressed``` uses ```createModuleConfig``` method which creates TreeWalker configuration. If I use the same method to create Checker configuration it will be the same logic as in the ```createCustomChecker``` now.

Reply:
> Explain what is the difference between the method which creates custom checker configuration with multiple children and custom verify method which will contain the logic which creates the checker with multiple children? 

Assignment to `Configuration` or `DefaultConfiguration` is locked to specific methods as we need to know which module we are instantiating and how. Adding a custom create method will hide what check and properties are being created in a separate method as we aren't traversing methods and following variable/parameter assignments.
`verify` method isn't as strictly checked, but we verify it exists so we know we are dealing with regression area.

>  I still need the checker instance which configuration consists of more than one child.

You don't need the instance for this.
Call `createRootConfig` with the parameter of the first child (or `null` if no child is needed), then call `addChild` for each extra child you wish to add.
Example: https://github.com/checkstyle/checkstyle/blob/ccd0421da49f4884afd576d31365ddc649218e9c/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWarningsFilterTest.java#L165-L169

We don't restrict working with Checker directly now, but it shouldn't be often you work with `Checker` directly unless it is for a `Checker` specific test.

> createModuleConfig(Checker.class);

You should use `createRootConfig` instead of `Checker.class` as we have special code to recognize the `Checker` module if you pass the verify method the `Configuration`.

> Did you mean that the verifySuppressed method should become:

It should become what you need, that is one example. `SuppressWarningsFilterTest` was able to use the same modules for each test as it was able to keep all messages in `ALL_MESSAGES` and then remove ones that were being suppressed in the `verifySuppressed` method.

Reply:
@rnveach 
Ok, I got it. Refactor my code. Please, review once again.